index.js

```diff
@@ -172,4 +172,115 @@ async function getFileHash(file) {
   })
 }
 
-main().catch(e => console.error(e))
+function addScriptToPackage(script) {
+  let pkg = getPackageJson()
+  if (!pkg.scripts) {
+    pkg.scripts = {}
+  }
+
+  const msg = `[relative-deps] Adding relative-deps to ${script} script in package.json`
+
+  if (!pkg.scripts[script]) {
+    console.log(msg);
+    pkg.scripts[script] = "relative-deps"
+
+  } else if(!pkg.scripts[script].includes("relative-deps")) {
+    console.log(msg);
+    pkg.scripts[script] = `${pkg.scripts[script]} && relative-deps`
+  }
+  setPackageData(pkg)
+}
+
+function installRelativeDepsPackage() {
+  let pkg = getPackageJson()
+
+  if (!(
+    (pkg.devDependencies && pkg.devDependencies["relative-deps"]) ||
+    (pkg.dependencies && pkg.dependencies["relative-deps"])
+  )) {
+    console.log('[relative-deps] Installing relative-deps package')
+    child_process.execSync(`${yarnCmd} add -D relative-deps`)
+  }
+}
+
+function setupEmptyRelativeDeps () {
+  let pkg = getPackageJson()
+
+  if (!pkg.relativeDependencies) {
+    console.log(`[relative-deps] Setting up relativeDependencies section in package.json`)
+    pkg.relativeDependencies = {}
+    setPackageData(pkg)
+  }
+}
+
+function initRelativeDeps({ script }) {
+  installRelativeDepsPackage()
+  setupEmptyRelativeDeps()
+  addScriptToPackage(script)
+}
+
+async function addRelativeDeps({ paths, dev, script }) {
+  initRelativeDeps({ script })
+
+  if (!paths || paths.length === 0) {
+    console.log(`[relative-deps][WARN] no paths provided running ${script}`)
+    child_process.execSync(`${yarnCmd} ${script}`)
+    return
+  }
+  const libraries = paths.map(relPath => {
+    const libPackagePath = path.resolve(process.cwd(), relPath, "package.json")
+    if (!fs.existsSync(libPackagePath)) {
+      console.error(
+        `[relative-deps][ERROR] Failed to resolve dependency ${relPath}`
+      )
+      process.exit(1)
+    }
+
+    const libraryPackageJson = JSON.parse(fs.readFileSync(libPackagePath, "utf-8"))
+
+    return {
+      relPath,
+      name: libraryPackageJson.name,
+      version: libraryPackageJson.version
+    }
+  })
+
+  let pkg = getPackageJson()
+
+  const depsKey = dev ? "devDependencies" : "dependencies"
+  if (!pkg[depsKey]) pkg[depsKey] = {}
+
+  libraries.forEach(library => {
+    if (!pkg[depsKey][library.name]) {
+      // console.log()
+      child_process.execSync(
```

Comment:
Note that this one should be allowed to fail (with a warning), to be able to work with libraries that haven't been published yet. In that case I think it also shouldn't be added as the normal dependencies, but to relative dependencies only

Reply:
Good point! Added this